pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -399,11 +437,51 @@ func buildInboundTCPFilter(mesh *meshconfig.MeshConfig, attrs attributes, node *
 	}
 }
 
-func addServiceConfig(filterConfigs map[string]*types.Struct, config *mccpb.ServiceConfig) map[string]*types.Struct {
+func addServiceConfig(dir direction, push *model.PushContext, hostname model.Hostname, in *plugin.InputParams,
```

Comment:
nit: can you please not merge inbound and outbound configs into one function. They are quite different.

Reply:
change to two method: addServiceConfigForOutbound and addServiceConfigForInbound